test_program/block_test.c

```diff
@@ -0,0 +1,60 @@
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <inttypes.h>
+#include <cmockery/cmockery.h>
+extern int get_block(int pos);
+extern int init_block_store();
+extern int insert_block(char *buffer,size_t length);
+extern int fini_block_store();
+static void test_return_values(void **state) 
+{ 
+        
+        assert_int_equal(init_block_store(), 0);
+
+}
+
+/* Test case that fails as leak_memory() leaks a dynamically allocated block.*/
+static void leak_memory_test(void **state) 
+{
+        
+        get_block(5);
```

Comment:
Hi luis,

A gentle reminder about the doubt we had asked earlier.
We had a doubt about what a setup state is,setup any expected mocks is.

Regards,
Team_es

On Fri, May 8, 2015 at 11:56 AM, josephaug26 <notifications@github.com>
wrote:

> In test_program/block_test.c
> <https://github.com/YADL/yadl/pull/12#discussion_r29919168>:
>
> > +extern int get_block(int pos);
> > +extern int init_block_store();
> > +extern int insert_block(char *buffer,size_t length);
> > +extern int fini_block_store();
> > +static void test_return_values(void **state)
> > +{
> > +
> > +        assert_int_equal(init_block_store(), 0);
> > +
> > +}
> > +
> > +/* Test case that fails as leak_memory() leaks a dynamically allocated block.*/
> > +static void leak_memory_test(void **state)
> > +{
> > +
> > +        get_block(5);
>
> Ewen/Srinivas can we have this fixed asap. If u have a doubt please ask
> Luis about it.
> Also this patch is old and will have merge conflicts so could u please fix
> this on a new patch.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/YADL/yadl/pull/12/files#r29919168>.
>